File: app/src/main/java/com/example/monappliimmo/HomeFragment.java
package com.example.monappliimmo;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class HomeFragment extends Fragment {

    private RecyclerView recyclerView;
    private AppartementAdapter adapter;

    // Cette méthode remplace le "setContentView" de l'activité
    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        // On relie le Java au fichier fragment_home.xml
        View view = inflater.inflate(R.layout.fragment_home, container, false);

        // On initialise le RecyclerView
        recyclerView = view.findViewById(R.id.recyclerViewHome);
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));

        // On lance le chargement des données
        chargerAppartements();

        return view;
    }

    private void chargerAppartements() {
        // On appelle ton API (assure-toi que RetrofitClient est bien configuré)
        RetrofitClient.getInstance().getApi().getAppartements().enqueue(new Callback<List<Appartement>>() {
            @Override
            public void onResponse(Call<List<Appartement>> call, Response<List<Appartement>> response) {
                if (response.isSuccessful() && response.body() != null) {
                    adapter = new AppartementAdapter(response.body(), null);
                    recyclerView.setAdapter(adapter);
                }
            }

            @Override
            public void onFailure(Call<List<Appartement>> call, Throwable t) {
                // Optionnel : ajouter un log d'erreur ici
            }
        });
    }
}

File: app/src/main/java/com/example/monappliimmo/HomeFragment.kt
package com.example.monappliimmo

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class HomeFragment : Fragment() {

    // On utilise lateinit car on sait qu'on va l'initialiser dans onCreateView
    private lateinit var recyclerView: RecyclerView
    private var adapter: AppartementAdapter? = null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Chargement du layout
        val view = inflater.inflate(R.layout.fragment_home, container, false)

        // Initialisation de la vue
        recyclerView = view.findViewById(R.id.recyclerViewHome)
        recyclerView.layoutManager = LinearLayoutManager(requireContext())

        chargerAppartements()

        return view
    }

    private fun chargerAppartements() {
        RetrofitClient.getInstance()
            .api // En Kotlin, on accède souvent aux getters comme des propriétés
            .getAppartements()
            .enqueue(object : Callback<List<Appartement>> {

                override fun onResponse(
                    call: Call<List<Appartement>>,
                    response: Response<List<Appartement>>
                ) {
                    if (response.isSuccessful) {
                        response.body()?.let { listAppart ->
                            // Mise à jour de l'UI
                            adapter = AppartementAdapter(listAppart)
                            recyclerView.adapter = adapter
                        }
                    }
                }

                override fun onFailure(call: Call<List<Appartement>>, t: Throwable) {
                    Log.e("HomeFragment", "Erreur API : ${t.message}")
                }
            })
    }
}